test/metadata_insertion_test.js

```diff
@@ -0,0 +1,99 @@
+const expect = require('chai').expect;
+
+describe('test-level metadata', () => {
+  // Original, no metadata
+  // it: string * fn -> fn 
+  it('should not appear on regular "it" tests', () => {
+      var test = it('should split on a delimiter', () => {
+          const parts = '1,2,3'.split(',');
+          expect(parts).to.eql(['1', '2', '3']);
+      });
+      expect(test.metadata).to.be.undefined;
+  });
+
+  // Metadata as 2nd parameter
+  // they: string * object * fn -> fn
+  it('should appear when specified as the 2nd parameter', () => {
+      var test = it('should split on a delimiter, with metadata as 2nd parameter', 
+        { requires: {topology: 'single'} }, () => {
+          const parts = '1,2,3'.split(',');
+          expect(parts).to.eql(['1', '2', '3']);
+      });
+      expect(test.metadata).to.eql({requires: {topology: 'single'}});
+  });
+
+
+  // Integra-style metadata
+  // theys: string * object -> fn
```

Comment:
looks like "theys" was left in by accident, probably will confuse readers in the future